src/SupervisionSuccess/model/produceProjections.js

```diff
@@ -98,6 +98,58 @@ function produceProjections(
     )
   );
 
+  const maxRevocationsByMonth = Array.from({
```

Comment:
These logical blocks, of which they are now 6 in this function, are all quite similar and could be extracted out to in-line helper functions. Something like this:
```
...
const months = BASE_YEAR + projections * 12 + 1;
const actualImplementationPeriod = BASE_YEAR + implementationPeriod;

function calculateRevocationsByMonth(revocationChange) {
  return Array.from({
    length: months + ADDED_MONTHS,
  }).map((revocations, month) =>
    calcRevocations(
      month,
      actualImplementationPeriod,
      revocationA,
      RAlpha0,
      revocationsTimescale,
      revocationChange
    )
  );
}

const revocationsByMonth = calculateRevocationsByMonth(changeInRevocations);
const maxRevocationsByMonth = calculateRevocationsByMonth(MAX_REVOCATION_CHANGE);
const minRevocationsByMonth = calculateRevocationsByMonth(MIN_REVOCATION_CHANGE);

function calculateNewAdmissionsByMonth(newAdmissionsChange) {
  return Array.from({
    length: months + ADDED_MONTHS,
  }).map((newOffense, month) =>
    calculateNewAdmissionsProjection(
      month,
      actualImplementationPeriod,
      newOffenseA,
      newOffenseAvgTimeServedInMonths,
      NAlpha0,
      newAdmissionsChange
    )
  );
}

const newAdmissionsByMonth = calculateNewAdmissionsByMonth(changeInNewAdmissions);
const maxNewAdmissionsByMonth = calculateNewAdmissionsByMonth(MAX_NEW_ADMISSION_CHANGE);
const minNewAdmissionsByMonth = calculateNewAdmissionsByMonth(MIN_NEW_ADMISSION_CHANGE);
```

Reply:
That's a good point, thanks!